### PR TITLE
Include create-app examples

### PR DESCRIPTION
**Pull request by tzyl**
This PR creates three rendered versions of the `@osdk/create-app` templates under the `examples/` folder where the existing examples were moved to `examples-extra/`.

- example-next-static-export
- example-react
- example-vue

These are almost exactly the raw output of the CLI apart from a few modifications:

- Swapped OSDK package to a locally generated package (reusing the 1.1 example)
- Removed .npmrc as we now use the locally generated package
- Replaced README explaining what the example actually is

With as few modifications as possible I'm hoping to next set up a script to automatically render these examples based on the template files and enforce that they're up to date so we know that the output of the templates successfully builds

### CODE REVIEW

File: packages/create-app/templates/template-next-static-export/tsconfig.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "compilerOptions": {
-    "target": "es5",
+    "target": "ES2020",
```
Comment (tzyl) [inline packages/create-app/templates/template-next-static-export/tsconfig.json:3]: This is to match the other templates based on vite which originally was needed when I had some of the generated code local to the example packages but might not matter anymore now that I reused an existing local package

File: packages/create-app/.gitignore
Comment (zeyadkhaled) [inline packages/create-app/.gitignore:1]: Why was this deleted?

File: examples/example-next-static-export/README.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# example-next-static-export
```
Comment (zeyadkhaled) [inline examples/example-next-static-export/README.md:1]: Do we want to add to the READMEs how to use/run these examples?

File: packages/create-app/.gitignore
Comment (tzyl) [reply packages/create-app/.gitignore:1]: Every version of the next template has `lib/` folders under `src/` that need to be remembered to be unignored which was getting annoying to manage so I changed the root .gitignore instead to allow `src/lib/` instead

File: examples/example-next-static-export/README.md (hunk 2 above)
Comment (tzyl) [reply examples/example-next-static-export/README.md:1]: Hmm, the thing is that there's only so much that you can do with the example because it needs a real ontology and stack to login and start loading data. I agree it would be nice if there was something more to see do you have any ideas? Ideally it stays as close as possible to the stock rendered template so we can keep it in sync automatically and it serves as our end to end test that the template is buildable

**Review by zeyadkhaled — APPROVED** [review]
